Cesium.Runtime/StdIoFunctions.cs

```diff
@@ -668,15 +667,66 @@ public static int Remove(byte* pathname)
         return 0;
     }
 
-    private static unsafe void* GetLastStream()
+    internal static StreamHandle? GetStream(void* filePtr)
     {
-        return (void*)(IntPtr)(Handles.Count - 1);
+        var handle = (IntPtr)filePtr;
+
+        var handleValue = handle.ToInt64();
+        if (handleValue is StdIn or StdOut or StdErr)
+        {
+            return Handles[(int)handleValue];
+        }
+
+        try
+        {
+            lock (_locker)
+            {
+                var gchAddr = Marshal.ReadIntPtr(handle);
+                var gch = GCHandle.FromIntPtr(gchAddr);
+
+                return gch.Target as StreamHandle;
+            }
```

Comment:
Hm, now wait, why we need this `_locker` at all? It doesn't seem to defend any collection manipulation. Let's maybe get rid of the `_locker`?